src/Microsoft.ML.Core/Data/DateTime.cs

```diff
@@ -201,7 +201,7 @@ public DvDateTimeZone(SysDateTimeOffset dto)
             // Since it is constructed from a SysDateTimeOffset, all the validations should work.
             var success = TryValidateOffset(dto.Offset.Ticks, out _offset);
             Contracts.Assert(success);
-            _dateTime = ValidateDate(new DvDateTime(dto.DateTime), ref _offset);
+            _dateTime = ValidateDate(new DvDateTime(dto.UtcDateTime), ref _offset);
```

Comment:
>new DvDateTime(dto.UtcDateTime), ref _offset); [](start = 37, length = 46)

Can we think about this particular one a bit more?

I don't know that this is correct, singe this `DateTimeZone` exists to encode a local time *including* a datetime. So I think that this change here might be inappropriate and is in fact introducing data corruption.